### PR TITLE
Sass color clean up

### PR DESCRIPTION
**Pull request by corykon**
Ready for some feedback. My apologies for touching so many files!

Couple notes:
- Let me know if I'm missing colors you know your app needs. Hopefully you'll find that for most everything in your various apps, one of the colors included here gets you pretty close. 
- You'll notice some tweaks to the button-component. I ripped out the hover/pressed state color variables in favor of just using a shade() function in a button styling mixin. Also, it made sense to change the names for the button color types based on the new variable structure.

### CODE REVIEW

File: src/app/lib/button/button-demo/button-demo.component.html [hunk 1]
```diff
@@ -7,17 +7,17 @@ <h5>
     </h5>
     <div class="sample-buttons">
         <button hc-button title="Primary Button" color="primary">Primary</button>
-        <button hc-button title="Primary-Alt1 Button" color="primary-alt1">Primary Alt1</button>
-        <button hc-button title="Primary-Alt2 Button" color="primary-alt2">Primary Alt2</button>
-        <button hc-button title="Primary-Alt3 Button" color="primary-alt3" >Primary Alt3</button>
+        <button hc-button title="Primary-Alt Button" color="primary-alt">Primary Alt</button>
+        <button hc-button title="Destructive Button" color="destructive">Destructive</button>
```
Comment (corytak) [inline src/app/lib/button/button-demo/button-demo.component.html]: I like the descriptive labels for the different primary button colors. It makes it seem less like a whimsical choice and more like the choice should be informed by the current context.

File: src/app/lib/button/button-demo/button-demo.component.html [hunk 2]
```diff
@@ -7,17 +7,17 @@ <h5>
     </h5>
     <div class="sample-buttons">
         <button hc-button title="Primary Button" color="primary">Primary</button>
-        <button hc-button title="Primary-Alt1 Button" color="primary-alt1">Primary Alt1</button>
-        <button hc-button title="Primary-Alt2 Button" color="primary-alt2">Primary Alt2</button>
-        <button hc-button title="Primary-Alt3 Button" color="primary-alt3" >Primary Alt3</button>
+        <button hc-button title="Primary-Alt Button" color="primary-alt">Primary Alt</button>
```
Comment (corytak) [inline src/app/lib/button/button-demo/button-demo.component.html]: I wonder if there's a similarly descriptive name to replace `Primary-Alt`

Comment (corykon) [reply src/app/lib/button/button-demo/button-demo.component.html]: Yeah, there may be. Any suggestions? I think there's likely another round of refactoring to come for this component soon, so I'll likely leave as is for now.

File: src/app/lib/color-demo/color-demo.component.html [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            .my-component &#123;
+                background-color: darken($slate-gray-100, 5%);
+                color: tint($dark-blue, 5%);
+            &#125;
+            </code>
+            </pre>
+        
+        
+        </li>
+        <li>If that doesn't suffice, or you are not using the sass variables, try using <a hc-button href="coolors.co" target="_blank">coolors.co</a>
```
Comment (andrew-frueh) [inline src/app/lib/color-demo/color-demo.component.html]: This is really minor @corykon - put make sure and add http:// to your link to coolors.co so it doesn't try and link to our cashmere site.

Comment (corykon) [reply src/app/lib/color-demo/color-demo.component.html]: good catch! thank you

File: src/app/lib/color-demo/color-demo.component.html [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    <h2>Color</h2>
+    <p>
+        Following are the suggested colors for use within your application. Some expansion beyond the general palette is acceptable as long as the
+        additional colors are similar in value/saturation. It's recommended that you import the colors and leverage the sass variables.
+    </p>
+    
+    <h4>Primary Palette</h4>
+    <p>The primary colors used for branding, calls to action (buttons, links, etc), and other accents.</p>
+    <div class="palette-row primary">
+        <hc-swatch-demo-component hex="#00aeff" name="blue" [isLarge]="true"></hc-swatch-demo-component>
```
[the remark this answers is not in the document]
Comment (corykon) [reply src/app/lib/color-demo/color-demo.component.html]: Using sass extract sounds pretty cool. You could create an issue for that.

**Pull request merged**